cmd/dockerd/config_common_unix.go

```diff
@@ -15,6 +20,24 @@ var (
 	defaultExecRoot = "/var/run/docker"
 )
 
+func init() {
+	if !rootless.RunningWithNonRootUsername {
+		return
+	}
+	//  pam_systemd sets XDG_RUNTIME_DIR but not other dirs.
+	if xdgDataHome := os.Getenv("XDG_DATA_HOME"); xdgDataHome != "" {
+		dirs := strings.Split(xdgDataHome, ":")
+		defaultDataRoot = filepath.Join(dirs[0], "docker")
+	} else if home := os.Getenv("HOME"); home != "" {
+		defaultDataRoot = filepath.Join(home, ".local", "share", "docker")
```

Comment:
Should we use the `homedir` package for this?

```suggestion
		defaultDataRoot = filepath.Join(homedir.Get(), ".local", "share", "docker")
```

Reply:
done